src/main/java/com/stablekernel/standardlib/BlockingProgressFragment.java

```diff
@@ -11,17 +11,37 @@
 import android.view.View;
 import android.view.ViewGroup;
 import android.view.Window;
+import android.widget.TextView;
 
 import rx.Subscription;
 
 public class BlockingProgressFragment extends DialogFragment {
     public static final String TAG = BlockingProgressFragment.class.getSimpleName();
+    private static final String ARGS_PROGRESS_TEXT = "ARGS_PROGRESS_TEXT";
     private Subscription subscription;
+    private String progressText;
 
     public static BlockingProgressFragment newInstance() {
         return new BlockingProgressFragment();
     }
 
+    public static BlockingProgressFragment newInstance(String progressText) {
+        BlockingProgressFragment fragment = new BlockingProgressFragment();
+        Bundle args = new Bundle();
+        args.putString(ARGS_PROGRESS_TEXT, progressText);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (getArguments() != null) {
```

Comment:
I would add to this:  `if (getArguments() != null && getArguments.containsKey(ARGS_PROGRESS_TEXT)`